Foundation/src/DateTime.cpp

```diff
@@ -43,14 +44,9 @@ DateTime::DateTime(const tm& tmStruct):
 	_millisecond(0),
 	_microsecond(0)
 {
-	poco_assert (_year >= 0 && _year <= 9999);
-	poco_assert (_month >= 1 && _month <= 12);
-	poco_assert (_day >= 1 && _day <= daysOfMonth(_year, _month));
-	poco_assert (_hour >= 0 && _hour <= 23);
-	poco_assert (_minute >= 0 && _minute <= 59);
-	poco_assert (_second >= 0 && _second <= 60);
-
-	_utcTime = toUtcTime(toJulianDay(_year, _month, _day)) + 10*(_hour*Timespan::HOURS + _minute*Timespan::MINUTES + _second*Timespan::SECONDS);
+	checkValid();
```

Comment:
Should `checkValid` be used after initialising `_utcTime`?

Reply:
`checkValid` does not check utc and it is pointless to calculate utc if something is invalid